From 52a38e42b894472515588b0de3d8f14b5ef642b9 Mon Sep 17 00:00:00 2001
From: Meriipu <Meriipu@users.noreply.github.com>
Date: Sat, 22 May 2021 15:38:47 +0200
Subject: [PATCH 05/10] do not auto-open urlbar on focus

---
 browser/components/urlbar/UrlbarInput.jsm | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/browser/components/urlbar/UrlbarInput.jsm b/browser/components/urlbar/UrlbarInput.jsm
index 395edee91f59..4fd2bd103a40 100644
--- a/browser/components/urlbar/UrlbarInput.jsm
+++ b/browser/components/urlbar/UrlbarInput.jsm
@@ -2848,7 +2848,8 @@ class UrlbarInput {
     }
 
     if (this.focusedViaMousedown) {
-      this.view.autoOpen({ event });
+      console.log("[UrlbarInput autoOpen focusedViaMousedown] haha no thanks");
+      //this.view.autoOpen({ event });
     } else if (this.inputField.hasAttribute("refocused-by-panel")) {
       this._maybeSelectAll(true); // not sure what true here does
     }
@@ -2915,6 +2916,8 @@ class UrlbarInput {
           this._preventClickSelectsAll = true;
           this.search(UrlbarTokenizer.RESTRICT.SEARCH);
         } else {
+          console.log("[UrlbarInput autoOpen] haha no thanks");
+          /*
           // Do not suppress the focus border if we are already focused. If we
           // did, we'd hide the focus border briefly then show it again if the
           // user has Top Sites disabled, creating a flashing effect.
@@ -2922,6 +2925,7 @@ class UrlbarInput {
             event,
             suppressFocusBorder: !this.hasAttribute("focused"),
           });
+          */
         }
         break;
       case this.dropmarker:
-- 
2.31.1

