From 7077b4e3246bc37f6805dc4d196d14ccec6cc487 Mon Sep 17 00:00:00 2001
From: Meriipu <Meriipu@users.noreply.github.com>
Date: Mon, 14 Dec 2020 06:31:25 +0100
Subject: [PATCH 5/8] do not auto-open urlbar on focus

---
 browser/components/urlbar/UrlbarInput.jsm | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/browser/components/urlbar/UrlbarInput.jsm b/browser/components/urlbar/UrlbarInput.jsm
index 989b1611afb1..f7f16d778b17 100644
--- a/browser/components/urlbar/UrlbarInput.jsm
+++ b/browser/components/urlbar/UrlbarInput.jsm
@@ -2754,7 +2754,8 @@ class UrlbarInput {
     this.startLayoutExtend();
 
     if (this.focusedViaMousedown) {
-      this.view.autoOpen({ event });
+      console.log("[UrlbarInput autoOpen focusedViaMousedown] haha no thanks");
+      //this.view.autoOpen({ event });
     } else if (this.inputField.hasAttribute("refocused-by-panel")) {
       this._maybeSelectAll(true); // not sure what true here does
     }
@@ -2821,7 +2822,8 @@ class UrlbarInput {
           this._preventClickSelectsAll = true;
           this.search(UrlbarTokenizer.RESTRICT.SEARCH);
         } else {
-          this.view.autoOpen({ event });
+          console.log("[UrlbarInput autoOpen] haha no thanks");
+          //this.view.autoOpen({ event });
         }
         break;
       case this.dropmarker:
-- 
2.29.2

