From e1aac65c8b49f5f69af74cc15958a5d74dc6b02e Mon Sep 17 00:00:00 2001
From: Meriipu <Meriipu@users.noreply.github.com>
Date: Fri, 17 Dec 2021 16:01:07 +0100
Subject: [PATCH 05/10] do not auto-open urlbar on focus

---
 browser/components/urlbar/UrlbarInput.jsm | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/browser/components/urlbar/UrlbarInput.jsm b/browser/components/urlbar/UrlbarInput.jsm
index 33130d51054a..2f6bbc481115 100644
--- a/browser/components/urlbar/UrlbarInput.jsm
+++ b/browser/components/urlbar/UrlbarInput.jsm
@@ -2828,7 +2828,8 @@ class UrlbarInput {
     }
 
     if (this.focusedViaMousedown) {
-      this.view.autoOpen({ event });
+      console.log("[UrlbarInput autoOpen focusedViaMousedown] haha no thanks");
+      //this.view.autoOpen({ event });
     } else if (this.inputField.hasAttribute("refocused-by-panel")) {
       this._maybeSelectAll(true); // not sure what true here does
     }
@@ -2898,6 +2899,8 @@ class UrlbarInput {
           this._preventClickSelectsAll = true;
           this.search(UrlbarTokenizer.RESTRICT.SEARCH);
         } else {
+          console.log("[UrlbarInput autoOpen] haha no thanks");
+          /*
           // Do not suppress the focus border if we are already focused. If we
           // did, we'd hide the focus border briefly then show it again if the
           // user has Top Sites disabled, creating a flashing effect.
@@ -2905,6 +2908,7 @@ class UrlbarInput {
             event,
             suppressFocusBorder: !hasFocus,
           });
+          */
         }
         break;
       case this.dropmarker:
-- 
2.34.1

