From 86a907673f95a276acc9d107bd95c7aff11fb64d Mon Sep 17 00:00:00 2001
From: Meriipu <Meriipu@users.noreply.github.com>
Date: Fri, 17 Dec 2021 16:02:39 +0100
Subject: [PATCH 08/10] Disable interventions (without using policies.json)

---
 browser/components/urlbar/UrlbarProviderInterventions.sys.mjs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/browser/components/urlbar/UrlbarProviderInterventions.sys.mjs b/browser/components/urlbar/UrlbarProviderInterventions.sys.mjs
index 2c2b13d9cb47..c69afd1bbc8b 100644
--- a/browser/components/urlbar/UrlbarProviderInterventions.sys.mjs
+++ b/browser/components/urlbar/UrlbarProviderInterventions.sys.mjs
@@ -500,6 +500,8 @@ class ProviderInterventions extends UrlbarProvider {
    * @returns {boolean} Whether this provider should be invoked for the search.
    */
   isActive(queryContext) {
+    // Disable interventions without dealing with policies.json
+    return false;
     if (
       !queryContext.searchString ||
       queryContext.searchString.length > UrlbarUtils.MAX_TEXT_LENGTH ||
-- 
2.42.1

