From 0845d46a108930dfde1e59492c607774ebf164e2 Mon Sep 17 00:00:00 2001
From: Meriipu <Meriipu@users.noreply.github.com>
Date: Fri, 17 Dec 2021 14:52:32 +0100
Subject: [PATCH 09/10] Stop resetting search engines on update

---
 toolkit/components/search/SearchService.jsm | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/toolkit/components/search/SearchService.jsm b/toolkit/components/search/SearchService.jsm
index 837706453e32..7b36e47eaf09 100644
--- a/toolkit/components/search/SearchService.jsm
+++ b/toolkit/components/search/SearchService.jsm
@@ -362,6 +362,12 @@ SearchService.prototype = {
         await this.removeEngine(engine);
         engineRemoved = true;
       }
+      // Find a better way to fund your executive's extravagant lifestyles
+      // than breaking search engine configs of users every single update.
+      else if (engine.name != "DuckDuckGo") {
+        await this.removeEngine(engine);
+        engineRemoved = true;
+      }
     }
     // If we've removed an engine, and we don't have any left, we need to
     // reload the engines - it is possible the settings just had one engine in it,
@@ -394,6 +400,12 @@ SearchService.prototype = {
     ) {
       return true;
     }
+
+    // Find a better way to fund your executive's extravagant lifestyles
+    // than breaking search engine configs of users every single update.
+    if (engine.name != "DuckDuckGo") {
+      return true;
+    }
     return false;
   },
 
-- 
2.34.1

