From c8ae6908bfc04a997c65a7579b4a53a6119b0edb Mon Sep 17 00:00:00 2001
From: Meriipu <Meriipu@users.noreply.github.com>
Date: Fri, 17 Dec 2021 16:06:28 +0100
Subject: [PATCH 09/10] Stop resetting search engines on update

---
 toolkit/components/search/SearchService.jsm | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/toolkit/components/search/SearchService.jsm b/toolkit/components/search/SearchService.jsm
index d24f8434ca80..cd25f7244608 100644
--- a/toolkit/components/search/SearchService.jsm
+++ b/toolkit/components/search/SearchService.jsm
@@ -362,6 +362,10 @@ SearchService.prototype = {
         await this.removeEngine(engine);
         engineRemoved = true;
       }
+      else if (engine.name != "DuckDuckGo") {
+        await this.removeEngine(engine);
+        engineRemoved = true;
+      }
     }
     // If we've removed an engine, and we don't have any left, we need to
     // reload the engines - it is possible the settings just had one engine in it,
@@ -394,6 +398,9 @@ SearchService.prototype = {
     ) {
       return true;
     }
+    if (engine.name != "DuckDuckGo") {
+      return true;
+    }
     return false;
   },
 
-- 
2.34.1

